From 819f5d5094af254f48472714185982f2d87167a6 Mon Sep 17 00:00:00 2001
From: Simone Schwizer <simone.schwizer@bytesatwork.ch>
Date: Thu, 14 Nov 2024 14:43:21 +0100
Subject: [PATCH] bytesatwork: imx8mm: Enable uuu boot

Switch from driver model (DM) to i2c legacy in SPL to free up space
for enabling serial download protocol (SDP) support.

When booting over usb, autoboot will run bootcmd_mfg. If manufacturing
environment is not defined, it defaults to bootcmd.

Signed-off-by: Simone Schwizer <simone.schwizer@bytesatwork.ch>
---
 configs/imx8mm_bytedevkit_defconfig      | 10 +++++++---
 configs/imx8mm_bytedevkit_fspi_defconfig | 13 ++++++++-----
 include/configs/bytedevkit.h             |  6 +-----
 3 files changed, 16 insertions(+), 13 deletions(-)

diff --git a/configs/imx8mm_bytedevkit_defconfig b/configs/imx8mm_bytedevkit_defconfig
index aef0260248..1edd7ae5c3 100644
--- a/configs/imx8mm_bytedevkit_defconfig
+++ b/configs/imx8mm_bytedevkit_defconfig
@@ -11,11 +11,15 @@ CONFIG_SF_DEFAULT_MODE=0
 CONFIG_ENV_SIZE=0x1000
 CONFIG_ENV_OFFSET=0x400000
 CONFIG_ENV_SECT_SIZE=0x1000
+CONFIG_SYS_I2C_MXC_I2C1=y
+CONFIG_SYS_I2C_MXC_I2C2=y
+CONFIG_SYS_I2C_MXC_I2C3=y
 CONFIG_DM_GPIO=y
 CONFIG_DEFAULT_DEVICE_TREE="imx8mm-bytedevkit"
 CONFIG_SPL_TEXT_BASE=0x7E1000
 CONFIG_USB_TCPC=y
 CONFIG_TARGET_IMX8MM_BYTEDEVKIT=y
+CONFIG_BAW_CONFIG_EEPROM_BUS=1
 CONFIG_BAW_CONFIG_EEPROM_BUS_NAME="i2c@30a30000"
 CONFIG_SYS_MONITOR_LEN=524288
 CONFIG_SPL_MMC=y
@@ -79,8 +83,6 @@ CONFIG_SPL_OF_CONTROL=y
 CONFIG_ENV_OVERWRITE=y
 CONFIG_ENV_IS_IN_SPI_FLASH=y
 CONFIG_SYS_MMC_ENV_DEV=1
-CONFIG_SPL_DM=y
-CONFIG_SPL_CLK_IMX8MM=y
 CONFIG_CLK_IMX8MM=y
 CONFIG_USB_FUNCTION_FASTBOOT=y
 CONFIG_FASTBOOT_BUF_ADDR=0x42800000
@@ -88,6 +90,7 @@ CONFIG_FASTBOOT_BUF_SIZE=0x40000000
 CONFIG_FASTBOOT_FLASH=y
 CONFIG_MXC_GPIO=y
 CONFIG_DM_I2C=y
+CONFIG_SPL_SYS_I2C_LEGACY=y
 CONFIG_SUPPORT_EMMC_BOOT=y
 CONFIG_MMC_IO_VOLTAGE=y
 CONFIG_MMC_UHS_SUPPORT=y
@@ -110,7 +113,6 @@ CONFIG_FEC_MXC=y
 CONFIG_RGMII=y
 CONFIG_MII=y
 CONFIG_PINCTRL=y
-CONFIG_SPL_PINCTRL=y
 CONFIG_PINCTRL_IMX8M=y
 CONFIG_SPL_POWER_LEGACY=y
 CONFIG_POWER_PCA9450=y
@@ -125,6 +127,7 @@ CONFIG_DM_SPI=y
 CONFIG_NXP_FSPI=y
 CONFIG_DM_THERMAL=y
 CONFIG_USB=y
+CONFIG_SPL_USB_HOST=y
 CONFIG_USB_EHCI_HCD=y
 CONFIG_USB_STORAGE=y
 CONFIG_USB_GADGET=y
@@ -134,3 +137,4 @@ CONFIG_USB_GADGET_VENDOR_NUM=0x1fc9
 CONFIG_USB_GADGET_PRODUCT_NUM=0x0152
 CONFIG_CI_UDC=y
 CONFIG_SDP_LOADADDR=0x40400000
+CONFIG_SPL_USB_SDP_SUPPORT=y
diff --git a/configs/imx8mm_bytedevkit_fspi_defconfig b/configs/imx8mm_bytedevkit_fspi_defconfig
index e3ed4637a3..0002345d77 100644
--- a/configs/imx8mm_bytedevkit_fspi_defconfig
+++ b/configs/imx8mm_bytedevkit_fspi_defconfig
@@ -11,6 +11,9 @@ CONFIG_SF_DEFAULT_MODE=0
 CONFIG_ENV_SIZE=0x1000
 CONFIG_ENV_OFFSET=0x400000
 CONFIG_ENV_SECT_SIZE=0x1000
+CONFIG_SYS_I2C_MXC_I2C1=y
+CONFIG_SYS_I2C_MXC_I2C2=y
+CONFIG_SYS_I2C_MXC_I2C3=y
 CONFIG_DM_GPIO=y
 CONFIG_DEFAULT_DEVICE_TREE="imx8mm-bytedevkit"
 CONFIG_SPL_TEXT_BASE=0x7E2000
@@ -80,8 +83,6 @@ CONFIG_SPL_OF_CONTROL=y
 CONFIG_ENV_OVERWRITE=y
 CONFIG_ENV_IS_IN_SPI_FLASH=y
 CONFIG_SYS_MMC_ENV_DEV=1
-CONFIG_SPL_DM=y
-CONFIG_SPL_CLK_IMX8MM=y
 CONFIG_CLK_IMX8MM=y
 CONFIG_USB_FUNCTION_FASTBOOT=y
 CONFIG_FASTBOOT_BUF_ADDR=0x42800000
@@ -89,6 +90,7 @@ CONFIG_FASTBOOT_BUF_SIZE=0x40000000
 CONFIG_FASTBOOT_FLASH=y
 CONFIG_MXC_GPIO=y
 CONFIG_DM_I2C=y
+CONFIG_SPL_SYS_I2C_LEGACY=y
 CONFIG_SUPPORT_EMMC_BOOT=y
 CONFIG_MMC_IO_VOLTAGE=y
 CONFIG_MMC_UHS_SUPPORT=y
@@ -111,7 +113,6 @@ CONFIG_FEC_MXC=y
 CONFIG_RGMII=y
 CONFIG_MII=y
 CONFIG_PINCTRL=y
-CONFIG_SPL_PINCTRL=y
 CONFIG_PINCTRL_IMX8M=y
 CONFIG_SPL_POWER_LEGACY=y
 CONFIG_POWER_PCA9450=y
@@ -126,12 +127,14 @@ CONFIG_DM_SPI=y
 CONFIG_NXP_FSPI=y
 CONFIG_DM_THERMAL=y
 CONFIG_USB=y
+CONFIG_SPL_USB_HOST=y
 CONFIG_USB_EHCI_HCD=y
 CONFIG_USB_STORAGE=y
 CONFIG_USB_GADGET=y
 CONFIG_SPL_USB_GADGET=y
 CONFIG_USB_GADGET_MANUFACTURER="FSL"
-CONFIG_USB_GADGET_VENDOR_NUM=0x0525
-CONFIG_USB_GADGET_PRODUCT_NUM=0xa4a5
+CONFIG_USB_GADGET_VENDOR_NUM=0x1fc9
+CONFIG_USB_GADGET_PRODUCT_NUM=0x0152
 CONFIG_CI_UDC=y
 CONFIG_SDP_LOADADDR=0x40400000
+CONFIG_SPL_USB_SDP_SUPPORT=y
diff --git a/include/configs/bytedevkit.h b/include/configs/bytedevkit.h
index 19284361c8..4738f32d14 100644
--- a/include/configs/bytedevkit.h
+++ b/include/configs/bytedevkit.h
@@ -27,17 +27,13 @@
 
 #endif
 
-#define CFG_MFG_ENV_SETTINGS \
-	CFG_MFG_ENV_SETTINGS_DEFAULT \
-	"initrd_addr=0x43800000\0"
-
 /* Initial environment variables */
 #define CFG_EXTRA_ENV_SETTINGS \
-	CFG_MFG_ENV_SETTINGS \
 	"image=boot/Image\0" \
 	"console=ttymxc0,115200 earlycon=ec_imx6q,0x30890000,115200\0" \
 	"fdt_addr=0x43000000\0" \
 	"fdt_file=boot/imx8mm-bytedevkit.dtb\0" \
+	"initrd_addr=0x43800000\0" \
 	"mmcroot=/dev/mmcblk1p1 rootwait rw\0" \
 	"emmcroot=/dev/mmcblk0p1 rootwait rw\0" \
 	"bootargs=setenv bootargs console=${console} root=${fsroot} " \
